src/webgl/p5.RendererGL.js

```diff
@@ -83,6 +81,93 @@ p5.RendererGL.prototype._initContext = function() {
     throw new Error(er);
   }
 };
+
+//This is helper function to reset the context anytime the attributes
+//are changed with setAttributes()
+
+p5.RendererGL.prototype._resetContext = function(attr, options, callback) {
+  var w = this.width;
+  var h = this.height;
+  var defaultId = 'defaultCanvas0';
+  var c = document.getElementById(defaultId);
+  if(c){
+    c.parentNode.removeChild(c);
+  }
+  c = document.createElement('canvas');
+  c.id = defaultId;
+  if (this._pInst._userNode) {
+    this._pInst._userNode.appendChild(c);
+  } else {
+    document.body.appendChild(c);
```

Comment:
That is my understanding but maybe I could be missing some edge case. This is how the rest of p5 handles it so I made assumptions.